src/graph/optimizer/rule/PushLimitDownIndexScanRule.cpp

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "graph/optimizer/rule/PushLimitDownIndexScanRule.h"
+
+#include "common/expression/BinaryExpression.h"
+#include "common/expression/ConstantExpression.h"
+#include "common/expression/Expression.h"
+#include "common/expression/FunctionCallExpression.h"
+#include "common/expression/LogicalExpression.h"
+#include "common/expression/UnaryExpression.h"
+#include "graph/optimizer/OptContext.h"
+#include "graph/optimizer/OptGroup.h"
+#include "graph/planner/plan/PlanNode.h"
+#include "graph/planner/plan/Query.h"
+#include "graph/visitor/ExtractFilterExprVisitor.h"
+
+using nebula::graph::IndexScan;
+using nebula::graph::Limit;
+using nebula::graph::PlanNode;
+using nebula::graph::Project;
+using nebula::graph::QueryContext;
+
+namespace nebula {
+namespace opt {
+
+std::unique_ptr<OptRule> PushLimitDownIndexScanRule::kInstance =
+    std::unique_ptr<PushLimitDownIndexScanRule>(new PushLimitDownIndexScanRule());
+
+PushLimitDownIndexScanRule::PushLimitDownIndexScanRule() { RuleSet::QueryRules().addRule(this); }
+
+const Pattern &PushLimitDownIndexScanRule::pattern() const {
+  static Pattern pattern = Pattern::create(
+      graph::PlanNode::Kind::kLimit,
+      {Pattern::create(graph::PlanNode::Kind::kProject,
+                       {Pattern::create(graph::PlanNode::Kind::kIndexScan, {}, true)})});
```

Comment:
here should not be one opt rule, but should be split into two rules: limit <-> project, limit -> indexscan

Reply:
Why need limit->indexscan?

Reply:
Push limit down indexscan

Reply:
The sentence only generate `limit`->`project`->`indexscan` plan

Reply:
You should consider the optimization rules from the perspective of the optimizer, not just for this case.

Reply:
We don't need to match the nonexistent pattern.